### PR TITLE
feat(core): Allow `bootstrapModule` to configure `NgZone` in providers

### PR DESCRIPTION
**Pull request by atscott**
See individual commits

### CODE REVIEW

**Comment by atscott** [comment]
[Green TGP](https://fusion2.corp.google.com/presubmit/654155630/OCL:654155630:BASE:654983280:1721702217937:c25bff28/targets)

**Review by alxhub — APPROVED** [review]
A+ refactoring.

File: packages/core/src/linker/ng_module_factory.ts [hunk 1]
```diff
@@ -7,7 +7,7 @@
  */
 
 import {Injector} from '../di/injector';
-import {EnvironmentInjector} from '../di/r3_injector';
+import {EnvironmentInjector, R3Injector} from '../di/r3_injector';
```
Comment (bougwal) [inline packages/core/src/linker/ng_module_factory.ts:10]: Q: Isn't this a leftover ? - I do not see where `R3Injector` used in this file

Comment (bougwal) [inline packages/core/src/linker/ng_module_factory.ts:10]: Q: Isn't this a leftover ? - I do not see where `R3Injector` is being used in this file 

```suggestion
import {EnvironmentInjector} from '../di/r3_injector';
```

**Comment by thePunderWoman** [comment]
This PR was merged into the repository by commit 8098945d7b703c335075a0efc09d058d45a8a181.

The changes were merged into the following branches: main

**Pull request closed without merging**